diff -up mplayer-export-2008-08-18/configure.dvdread mplayer-export-2008-08-18/configure
--- mplayer-export-2008-08-18/configure.dvdread	2008-08-18 00:02:05.000000000 +0200
+++ mplayer-export-2008-08-18/configure	2008-08-18 22:47:42.000000000 +0200
@@ -5458,10 +5458,10 @@ elif test "$_dvdread" = auto ; then
   if test "$_dl" = yes; then
     cat > $TMPC << EOF
 #include <inttypes.h>
-#include <libdvdread/dvd_reader.h>
-#include <libdvdread/ifo_types.h>
-#include <libdvdread/ifo_read.h>
-#include <libdvdread/nav_read.h>
+#include <dvdread/dvd_reader.h>
+#include <dvdread/ifo_types.h>
+#include <dvdread/ifo_read.h>
+#include <dvdread/nav_read.h>
 int main(void) { return 0; }
 EOF
   fi
diff -up mplayer-export-2008-08-18/stream/stream_dvd_common.c.dvdread mplayer-export-2008-08-18/stream/stream_dvd_common.c
--- mplayer-export-2008-08-18/stream/stream_dvd_common.c.dvdread	2008-07-30 14:01:30.000000000 +0200
+++ mplayer-export-2008-08-18/stream/stream_dvd_common.c	2008-08-18 22:49:46.000000000 +0200
@@ -1,10 +1,6 @@
 #include "config.h"
 #include <inttypes.h>
-#ifdef CONFIG_DVDREAD_INTERNAL
 #include <dvdread/ifo_types.h>
-#else
-#include <libdvdread/ifo_types.h>
-#endif
 #include "stream_dvd_common.h"
 
 /** 
diff -up mplayer-export-2008-08-18/stream/stream_dvd_common.h.dvdread mplayer-export-2008-08-18/stream/stream_dvd_common.h
--- mplayer-export-2008-08-18/stream/stream_dvd_common.h.dvdread	2008-07-30 14:01:30.000000000 +0200
+++ mplayer-export-2008-08-18/stream/stream_dvd_common.h	2008-08-18 22:50:11.000000000 +0200
@@ -3,11 +3,7 @@
 
 #include "config.h"
 #include <inttypes.h>
-#ifdef CONFIG_DVDREAD_INTERNAL
 #include <dvdread/ifo_types.h>
-#else
-#include <libdvdread/ifo_types.h>
-#endif
 
 int mp_dvdtimetomsec(dvd_time_t *dt);
 
diff -up mplayer-export-2008-08-18/stream/stream_dvd.h.dvdread mplayer-export-2008-08-18/stream/stream_dvd.h
--- mplayer-export-2008-08-18/stream/stream_dvd.h.dvdread	2008-07-30 14:01:30.000000000 +0200
+++ mplayer-export-2008-08-18/stream/stream_dvd.h	2008-08-18 22:50:47.000000000 +0200
@@ -9,10 +9,10 @@
 #include "dvdread/ifo_read.h"
 #include "dvdread/nav_read.h"
 #else
-#include <libdvdread/dvd_reader.h>
-#include <libdvdread/ifo_types.h>
-#include <libdvdread/ifo_read.h>
-#include <libdvdread/nav_read.h>
+#include <dvdread/dvd_reader.h>
+#include <dvdread/ifo_types.h>
+#include <dvdread/ifo_read.h>
+#include <dvdread/nav_read.h>
 #endif
 #include "stream.h"
 
