diff -up mplayer-export-2008-08-18/Makefile.manlinks mplayer-export-2008-08-18/Makefile
--- mplayer-export-2008-08-18/Makefile.manlinks	2008-08-14 22:42:54.000000000 +0200
+++ mplayer-export-2008-08-18/Makefile	2008-08-19 13:50:17.000000000 +0200
@@ -830,7 +830,7 @@ install-mencoder-man: $(foreach lang,$(M
 install-mplayer-man:  $(foreach lang,$(MAN_LANGS),install-mplayer-man-$(lang))
 
 install-mencoder-man-en: install-mplayer-man-en
-	cd $(MANDIR)/man1 && ln -sf mplayer.1 mencoder.1
+	echo ".so mplayer.1" > $(MANDIR)/man1/mencoder.1
 
 install-mplayer-man-en:
 	$(INSTALL) -d $(MANDIR)/man1
@@ -838,7 +838,7 @@ install-mplayer-man-en:
 
 define MENCODER_MAN_RULE
 install-mencoder-man-$(lang): install-mplayer-man-$(lang)
-	cd $(MANDIR)/$(lang)/man1 && ln -sf mplayer.1 mencoder.1
+	echo ".so mplayer.1" > $(MANDIR)/$(lang)/man1/mencoder.1
 endef
 
 define MPLAYER_MAN_RULE
