diff -up mplayer-export-2010-03-27/configure.libgif mplayer-export-2010-03-27/configure
--- mplayer-export-2010-03-27/configure.libgif	2010-03-27 22:56:17.000000000 +0100
+++ mplayer-export-2010-03-27/configure	2010-03-28 00:37:40.000000000 +0100
@@ -5246,7 +5246,7 @@ cat > $TMPC << EOF
 #include <gif_lib.h>
 int main(void) { return 0; }
 EOF
-  for _ld_gif in "-lungif" "-lgif" ; do
+  for _ld_gif in "-lgif" "-lungif" ; do
     cc_check $_ld_gif && _gif=yes && break
   done
 fi
