diff -up mplayer-export-2009-03-19/DOCS/xml/configure.htmldocs mplayer-export-2009-03-19/DOCS/xml/configure
--- mplayer-export-2009-03-19/DOCS/xml/configure.htmldocs	2009-03-17 12:47:27.000000000 +0100
+++ mplayer-export-2009-03-19/DOCS/xml/configure	2009-03-19 21:48:30.000000000 +0100
@@ -187,7 +187,7 @@ then
   then
     echo "Found xsltproc. If it works, it's probably the best choice."
     # HACK: xslt always need a target file if a doctype is set
-    _xsltcommand="OUTPUT=\"\$1\"; if test -d \"\$OUTPUT\" ; then OUTPUT=\"$OUTPUT/dummy.html\" ; fi;"
+    _xsltcommand="OUTPUT=\"\$1\"; if test -d \"\$OUTPUT\" ; then OUTPUT=\"\$OUTPUT/dummy.html\" ; fi;"
     if test -n "$_catalog"
     then
       _xsltcommand="$_xsltcommand xsltproc --catalogs -o \$OUTPUT \$2 \$3"
