diff -up mplayer-export-2013-01-23/configure.nodvdcss mplayer-export-2013-01-23/configure
--- mplayer-export-2013-01-23/configure.nodvdcss	2013-02-02 10:03:58.000000000 +0100
+++ mplayer-export-2013-01-23/configure	2013-02-03 16:35:16.537604524 +0100
@@ -780,7 +780,7 @@ _dvdnavconfig=dvdnav-config
 _dvdreadconfig=dvdread-config
 _dvdread=auto
 _dvdread_internal=auto
-_libdvdcss_internal=auto
+_libdvdcss_internal=no
 _xanim=auto
 _real=auto
 _live=auto
diff -up mplayer-export-2013-01-23/Makefile.nodvdcss mplayer-export-2013-01-23/Makefile
--- mplayer-export-2013-01-23/Makefile.nodvdcss	2013-02-03 16:35:16.533604493 +0100
+++ mplayer-export-2013-01-23/Makefile	2013-02-03 16:35:16.537604524 +0100
@@ -124,11 +124,6 @@ SRCS_COMMON-$(LIBDCA)                +=
 SRCS_COMMON-$(LIBDV)                 += libmpcodecs/ad_libdv.c \
                                         libmpcodecs/vd_libdv.c \
                                         libmpdemux/demux_rawdv.c
-SRCS_COMMON-$(LIBDVDCSS_INTERNAL)    += libdvdcss/css.c \
-                                        libdvdcss/device.c \
-                                        libdvdcss/error.c \
-                                        libdvdcss/ioctl.c \
-                                        libdvdcss/libdvdcss.c \
 
 SRCS_COMMON-$(LIBMAD)                += libmpcodecs/ad_libmad.c
 
@@ -720,7 +715,6 @@ DIRS =  . \
 
 ALL_DIRS = $(DIRS) \
         libass \
-        libdvdcss \
         libdvdnav \
         libdvdnav/vm \
         libdvdread4 \
@@ -847,7 +841,6 @@ osdep/mplayer-rc.o: osdep/mplayer.exe.ma
 
 gui/%: CFLAGS += -Wno-strict-prototypes
 
-libdvdcss/%:   CFLAGS := -Ilibdvdcss -D_GNU_SOURCE -DVERSION=\"1.2.12\" $(CFLAGS_LIBDVDCSS) $(CFLAGS)
 libdvdnav/%:   CFLAGS := -Ilibdvdnav -D_GNU_SOURCE -DHAVE_CONFIG_H -DVERSION=\"MPlayer-custom\" $(CFLAGS)
 libdvdread4/%: CFLAGS := -Ilibdvdread4 -D_GNU_SOURCE $(CFLAGS_LIBDVDCSS_DVDREAD) $(CFLAGS)
 
