Index: libmpdemux/demux_rtp_codec.cpp
===================================================================
--- libmpdemux/demux_rtp_codec.cpp	(revision 27448)
+++ libmpdemux/demux_rtp_codec.cpp	(revision 27449)
@@ -198,6 +198,7 @@
   sh_audio->wf = wf;
   demux_stream_t* d_audio = demuxer->audio;
   d_audio->sh = sh_audio; sh_audio->ds = d_audio;
+  d_audio->id = sh_audio->aid;
   
   wf->nChannels = subsession->numChannels();
 
