diff -up mplayer-export-2013-03-29/configure.nodvdcss mplayer-export-2013-03-29/configure
--- mplayer-export-2013-03-29/configure.nodvdcss	2013-03-28 08:45:33.000000000 +0100
+++ mplayer-export-2013-03-29/configure	2013-03-28 08:46:54.302523303 +0100
@@ -741,7 +741,7 @@ _dvdnavconfig=dvdnav-config
 _dvdreadconfig=dvdread-config
 _dvdread=auto
 _dvdread_internal=auto
-_libdvdcss_internal=auto
+_libdvdcss_internal=no
 _xanim=auto
 _real=auto
 _live=auto
diff -up mplayer-export-2013-03-29/Makefile.nodvdcss mplayer-export-2013-03-29/Makefile
--- mplayer-export-2013-03-29/Makefile.nodvdcss	2013-03-28 08:46:54.300523311 +0100
+++ mplayer-export-2013-03-29/Makefile	2013-03-28 08:48:09.386242474 +0100
@@ -124,11 +124,6 @@ SRCS_COMMON-$(LIBDCA)                +=
 SRCS_COMMON-$(LIBDV)                 += libmpcodecs/ad_libdv.c          \
                                         libmpcodecs/vd_libdv.c          \
                                         libmpdemux/demux_rawdv.c
-SRCS_COMMON-$(LIBDVDCSS_INTERNAL)    += libdvdcss/css.c                 \
-                                        libdvdcss/device.c              \
-                                        libdvdcss/error.c               \
-                                        libdvdcss/ioctl.c               \
-                                        libdvdcss/libdvdcss.c           \
 
 SRCS_COMMON-$(LIBMAD)                += libmpcodecs/ad_libmad.c
 
@@ -730,7 +725,6 @@ DIRS =  .                       \
 
 ALL_DIRS = $(DIRS)              \
            libass               \
-           libdvdcss            \
            libdvdnav            \
            libdvdnav/vm         \
            libdvdread4          \
@@ -859,7 +853,6 @@ osdep/mplayer-rc.o: osdep/mplayer.exe.ma
 
 gui/%: CFLAGS += -Wno-strict-prototypes
 
-libdvdcss/%:   CFLAGS := $(CFLAGS_LIBDVDCSS) $(CFLAGS)
 libdvdnav/%:   CFLAGS := -Ilibdvdnav -DHAVE_CONFIG_H -DVERSION=\"MPlayer-custom\" $(CFLAGS)
 libdvdread4/%: CFLAGS := -Ilibdvdread4 $(CFLAGS_LIBDVDCSS_DVDREAD) $(CFLAGS)
 
