diff -up mplayer-export-2009-02-04/libswscale/swscale.c.ppc-compile mplayer-export-2009-02-04/libswscale/swscale.c
--- mplayer-export-2009-02-04/libswscale/swscale.c.ppc-compile	2009-01-25 20:58:33.000000000 +0100
+++ mplayer-export-2009-02-04/libswscale/swscale.c	2009-02-13 15:00:57.000000000 +0100
@@ -955,13 +955,12 @@ static inline void yuv2rgbXinC_full(SwsC
 
 //Note: we have C, X86, MMX, MMX2, 3DNOW versions, there is no 3DNOW+MMX2 one
 //Plain C versions
-#if !HAVE_MMX || defined (RUNTIME_CPUDETECT) || !CONFIG_GPL
+#if !HAVE_MMX || !HAVE_ALTIVEC || defined (RUNTIME_CPUDETECT) || !CONFIG_GPL
 #define COMPILE_C
 #endif
 
 #if ARCH_PPC
 #if (HAVE_ALTIVEC || defined (RUNTIME_CPUDETECT)) && CONFIG_GPL
-#undef COMPILE_C
 #define COMPILE_ALTIVEC
 #endif
 #endif //ARCH_PPC
