diff -up mplayer-export-2008-09-03/libvo/x11_common.c.bs mplayer-export-2008-09-03/libvo/x11_common.c
--- mplayer-export-2008-09-03/libvo/x11_common.c.bs	2008-08-04 08:16:23.000000000 +0200
+++ mplayer-export-2008-09-03/libvo/x11_common.c	2008-11-23 21:39:27.000000000 +0100
@@ -1250,7 +1250,7 @@ Window vo_x11_create_smooth_window(Displ
                                    unsigned int width, unsigned int height,
                                    int depth, Colormap col_map)
 {
-    unsigned long xswamask = CWBackingStore | CWBorderPixel;
+    unsigned long xswamask = CWBorderPixel;
     XSetWindowAttributes xswa;
     Window ret_win;
 
@@ -1261,7 +1261,7 @@ Window vo_x11_create_smooth_window(Displ
     }
     xswa.background_pixel = 0;
     xswa.border_pixel = 0;
-    xswa.backing_store = Always;
+    xswa.backing_store = NotUseful;
     xswa.bit_gravity = StaticGravity;
 
     ret_win =
