diff -up mplayer-export-2009-03-29/configure.cflags mplayer-export-2009-03-29/configure
--- mplayer-export-2009-03-29/configure.cflags	2009-03-29 00:21:21.000000000 +0100
+++ mplayer-export-2009-03-29/configure	2009-03-30 22:58:32.000000000 +0200
@@ -795,7 +795,7 @@ for ac_option do
     ;;
 
   --extra-cflags=*)
-    extra_cflags=`echo $ac_option | cut -d '=' -f 2`
+    extra_cflags=`echo $ac_option | cut -d '=' -f 2-`
     ;;
   --extra-ldflags=*)
     extra_ldflags=`echo $ac_option | cut -d '=' -f 2`
