diff -up mplayer-export-2009-01-07/libavcodec/Makefile.compile mplayer-export-2009-01-07/libavcodec/Makefile
--- mplayer-export-2009-01-07/libavcodec/Makefile.compile	2009-01-06 23:01:57.000000000 +0100
+++ mplayer-export-2009-01-07/libavcodec/Makefile	2009-01-08 21:32:58.000000000 +0100
@@ -511,3 +511,7 @@ include $(SUBDIR)../subdir.mak
 
 $(SUBDIR)dct-test$(EXESUF): $(SUBDIR)fdctref.o $(SUBDIR)aandcttab.o
 $(SUBDIR)fft-test$(EXESUF): $(SUBDIR)fdctref.o
+ifeq ($(ARCH_X86_32),yes)
+$(SUBDIR)x86/snowdsp_mmx.o: CFLAGS += -fomit-frame-pointer
+$(SUBDIR)h264.o: CFLAGS += -fomit-frame-pointer
+endif
