diff -up mplayer-export-2008-06-04/Makefile.manlinks mplayer-export-2008-06-04/Makefile
--- mplayer-export-2008-06-04/Makefile.manlinks	2008-05-30 22:10:36.000000000 +0200
+++ mplayer-export-2008-06-04/Makefile	2008-06-04 23:13:37.000000000 +0200
@@ -805,7 +805,7 @@ install-mplayer-man-en:
 	$(INSTALL) -c -m 644 DOCS/man/en/mplayer.1 $(MANDIR)/man1/
 
 install-mencoder-man-en: install-mplayer-man-en
-	cd $(MANDIR)/man1 && ln -sf mplayer.1 mencoder.1
+	echo ".so mplayer.1" > $(MANDIR)/man1/mencoder.1
 
 define MPLAYER_MAN_RULE
 install-mplayer-man-$(lang):
@@ -815,7 +815,7 @@ endef
 
 define MENCODER_MAN_RULE
 install-mencoder-man-$(lang): install-mplayer-man-$(lang)
-	cd $(MANDIR)/$(lang)/man1 && ln -sf mplayer.1 mencoder.1
+	echo ".so mplayer.1" > $(MANDIR)/$(lang)/man1/mencoder.1
 endef
 
 $(foreach lang,$(MAN_LANG),$(eval $(MPLAYER_MAN_RULE)))
