diff -up mplayer-export-2011-08-16/configure.nodvdcss mplayer-export-2011-08-16/configure
--- mplayer-export-2011-08-16/configure.nodvdcss	2011-08-11 19:45:43.000000000 +0200
+++ mplayer-export-2011-08-16/configure	2011-08-16 21:53:23.000000000 +0200
@@ -731,7 +731,7 @@ _dvdnavconfig=dvdnav-config
 _dvdreadconfig=dvdread-config
 _dvdread=auto
 _dvdread_internal=auto
-_libdvdcss_internal=auto
+_libdvdcss_internal=no
 _xanim=auto
 _real=auto
 _live=auto
diff -up mplayer-export-2011-08-16/Makefile.nodvdcss mplayer-export-2011-08-16/Makefile
--- mplayer-export-2011-08-16/Makefile.nodvdcss	2011-08-16 21:53:23.000000000 +0200
+++ mplayer-export-2011-08-16/Makefile	2011-08-16 21:53:55.000000000 +0200
@@ -112,11 +112,6 @@ SRCS_COMMON-$(LIBDCA)                += 
 SRCS_COMMON-$(LIBDV)                 += libmpcodecs/ad_libdv.c \
                                         libmpcodecs/vd_libdv.c \
                                         libmpdemux/demux_rawdv.c
-SRCS_COMMON-$(LIBDVDCSS_INTERNAL)    += libdvdcss/css.c \
-                                        libdvdcss/device.c \
-                                        libdvdcss/error.c \
-                                        libdvdcss/ioctl.c \
-                                        libdvdcss/libdvdcss.c \
 
 SRCS_COMMON-$(LIBMAD)                += libmpcodecs/ad_libmad.c
 
@@ -741,7 +736,6 @@ DIRS =  . \
 
 ALL_DIRS = $(DIRS) \
         libass \
-        libdvdcss \
         libdvdnav \
         libdvdnav/vm \
         libdvdread4 \
@@ -830,7 +824,6 @@ osdep/mplayer-rc.o: osdep/mplayer.exe.ma
 
 gui/%: CFLAGS += -Wno-strict-prototypes
 
-libdvdcss/%:   CFLAGS := -Ilibdvdcss -D_GNU_SOURCE -DVERSION=\"1.2.10\" $(CFLAGS_LIBDVDCSS) $(CFLAGS)
 libdvdnav/%:   CFLAGS := -Ilibdvdnav -D_GNU_SOURCE -DHAVE_CONFIG_H -DVERSION=\"MPlayer-custom\" $(CFLAGS)
 libdvdread4/%: CFLAGS := -Ilibdvdread4 -D_GNU_SOURCE $(CFLAGS_LIBDVDCSS_DVDREAD) $(CFLAGS)
 
