diff -up mplayer-export-2010-12-04/configure.nodvdcss mplayer-export-2010-12-04/configure
--- mplayer-export-2010-12-04/configure.nodvdcss	2010-11-03 09:12:22.000000000 +0100
+++ mplayer-export-2010-12-04/configure	2010-12-05 00:01:29.000000000 +0100
@@ -683,7 +683,7 @@ _dvdnavconfig=dvdnav-config
 _dvdreadconfig=dvdread-config
 _dvdread=auto
 _dvdread_internal=auto
-_libdvdcss_internal=auto
+_libdvdcss_internal=no
 _xanim=auto
 _real=auto
 _live=auto
diff -up mplayer-export-2010-12-04/Makefile.nodvdcss mplayer-export-2010-12-04/Makefile
--- mplayer-export-2010-12-04/Makefile.nodvdcss	2010-12-05 00:01:29.000000000 +0100
+++ mplayer-export-2010-12-04/Makefile	2010-12-05 00:02:07.000000000 +0100
@@ -150,11 +150,6 @@ SRCS_COMMON-$(LIBDCA)                += 
 SRCS_COMMON-$(LIBDV)                 += libmpcodecs/ad_libdv.c \
                                         libmpcodecs/vd_libdv.c \
                                         libmpdemux/demux_rawdv.c
-SRCS_COMMON-$(LIBDVDCSS_INTERNAL)    += libdvdcss/css.c \
-                                        libdvdcss/device.c \
-                                        libdvdcss/error.c \
-                                        libdvdcss/ioctl.c \
-                                        libdvdcss/libdvdcss.c \
 
 SRCS_COMMON-$(LIBMAD)                += libmpcodecs/ad_libmad.c
 
@@ -781,7 +776,6 @@ DIRS =  . \
         libaf \
         libao2 \
         libass \
-        libdvdcss \
         libdvdnav \
         libdvdnav/vm \
         libdvdread4 \
@@ -898,7 +892,6 @@ osdep/mplayer-rc.o: osdep/mplayer.exe.ma
 
 gui/%: CFLAGS += -Wno-strict-prototypes
 
-libdvdcss/%:   CFLAGS := -Ilibdvdcss -D_GNU_SOURCE -DVERSION=\"1.2.10\" $(CFLAGS_LIBDVDCSS) $(CFLAGS)
 libdvdnav/%:   CFLAGS := -Ilibdvdnav -D_GNU_SOURCE -DHAVE_CONFIG_H -DVERSION=\"MPlayer-custom\" $(CFLAGS)
 libdvdread4/%: CFLAGS := -Ilibdvdread4 -D_GNU_SOURCE $(CFLAGS_LIBDVDCSS_DVDREAD) $(CFLAGS)
 libfaad2/%:    CFLAGS := -Ilibfaad2 -DHAVE_CONFIG_H $(CFLAGS_FAAD_FIXED) $(CFLAGS)
