diff -up mplayer-export-2010-12-04/configure.libvorbis mplayer-export-2010-12-04/configure
--- mplayer-export-2010-12-04/configure.libvorbis	2010-12-05 00:14:13.000000000 +0100
+++ mplayer-export-2010-12-04/configure	2010-12-05 00:16:53.000000000 +0100
@@ -659,6 +659,7 @@ _tremor=auto
 _tremor_internal=yes
 _tremor_low=no
 _libvorbis=auto
+_libvorbisenc=auto
 _speex=auto
 _libgsm=auto
 _theora=auto
@@ -6315,7 +6316,7 @@ if test "$_toolamedir" ; then
 fi
 echores "$_toolame"
 
-echocheck "OggVorbis support"
+echocheck "OggVorbis decoding support"
 if test "$_tremor_internal" = yes; then
   _libvorbis=no
 elif test "$_tremor" = auto; then
@@ -6328,6 +6329,7 @@ if test "$_libvorbis" = auto; then
 fi
 if test "$_tremor_internal" = yes ; then
   _vorbis=yes
+  def_libvorbis='#define CONFIG_LIBVORBIS 0'
   def_vorbis='#define CONFIG_OGGVORBIS 1'
   def_tremor='#define CONFIG_TREMOR 1'
   codecmodules="tremor(internal) $codecmodules"
@@ -6338,6 +6340,7 @@ if test "$_tremor_internal" = yes ; then
   fi
 elif test "$_tremor" = yes ; then
   _vorbis=yes
+  def_libvorbis='#define CONFIG_LIBVORBIS 0'
   def_vorbis='#define CONFIG_OGGVORBIS 1'
   def_tremor='#define CONFIG_TREMOR 1'
   codecmodules="tremor(external) $codecmodules"
@@ -6345,17 +6348,32 @@ elif test "$_tremor" = yes ; then
   extra_ldflags="$extra_ldflags -logg -lvorbisidec"
 elif test "$_libvorbis" = yes ; then
   _vorbis=yes
+  def_libvorbis='#define CONFIG_LIBVORBIS 1'
   def_vorbis='#define CONFIG_OGGVORBIS 1'
   codecmodules="libvorbis $codecmodules"
   res_comment="libvorbis"
   extra_ldflags="$extra_ldflags -lvorbis -logg"
-  libavencoders="$libavencoders LIBVORBIS_ENCODER"
 else
   _vorbis=no
   nocodecmodules="libvorbis $nocodecmodules"
 fi
 echores "$_vorbis"
 
+echocheck "OggVorbis encoding support"
+if test "$_libvorbisenc" = auto; then
+  _libvorbisenc=no
+  statement_check vorbis/vorbisenc.h 'vorbis_encode_init(NULL, 0, 0, 0, 0, 0)' -lvorbisenc -lvorbis -logg $_ld_lm && _libvorbisenc=yes
+fi
+if test "$_libvorbisenc" = yes ; then
+  _vorbisenc=yes
+  res_comment="libvorbisenc"
+  extra_ldflags="$extra_ldflags -lvorbisenc -lvorbis -logg"
+  libavencoders="$libavencoders LIBVORBIS_ENCODER"
+else
+  _vorbisenc=no
+fi
+echores "$_vorbisenc"
+
 echocheck "libspeex (version >= 1.1 required)"
 if test "$_speex" = auto ; then
   _speex=no
@@ -8725,7 +8743,7 @@ $def_yasm
 #define CONFIG_FFSERVER 0
 #define CONFIG_GPL 1
 #define CONFIG_GRAY 0
-#define CONFIG_LIBVORBIS 0
+$def_libvorbis
 #define CONFIG_POWERPC_PERF 0
 #define CONFIG_SHARED 0
 #define CONFIG_SMALL 0
