diff -up mplayer-export-2008-08-18/etc/example.conf.orig mplayer-export-2008-08-18/etc/example.conf
--- mplayer-export-2008-08-18/etc/example.conf.orig	2008-02-08 19:33:39.000000000 +0100
+++ mplayer-export-2008-08-18/etc/example.conf	2008-08-19 13:47:15.000000000 +0200
@@ -1,7 +1,7 @@
 #
 # MPlayer configuration file
 #
-# Configuration files are read system-wide from /usr/local/etc/mplayer.conf
+# Configuration files are read system-wide from /etc/mplayer/mplayer.conf
 # and per user from ~/.mplayer/config, where per-user settings override
 # system-wide settings, all of which are overrriden by the command line.
 #
@@ -71,7 +71,7 @@
 ##################
 
 # Specify default audio driver (see -ao help for a list).
-#ao=oss
+ao = pulse,alsa,
 
 # Use SDL audio driver with the esd subdriver by default.
 #ao = sdl:esd
@@ -95,7 +95,7 @@
 #framedrop = yes
 
 # Specify your preferred skin here (skins are searched for in
-# /usr/local/share/mplayer/skins/<name> and ~/.mplayer/skins/<name>).
+# /usr/share/mplayer/skins/<name> and ~/.mplayer/skins/<name>).
 #skin = Abyss
 
 # Resample the font alphamap.
@@ -105,6 +105,15 @@
 # 10    bold black outline
 #ffactor = 0.75
 
+# FriBidi can do "funny" things to some ISO-8859-* subtitles
+flip-hebrew = no
+
+# Use fontconfig for font selection
+fontconfig = yes
+
+# Default font for unstyled subtitles
+font = "Sans"
+
 # cache settings
 #
 # Use 8MB input cache by default.
@@ -122,6 +131,9 @@
 # DVD: Play English audio tracks if available.
 #alang = en
 
+# A plugged in joystick can interfere with keyboard shortcuts
+joystick = no
+
 ###################
 # DVDNAV Settings #
 ###################
