diff -up mplayer-export-2008-09-03/configure.nodvdcss mplayer-export-2008-09-03/configure
--- mplayer-export-2008-09-03/configure.nodvdcss	2008-10-16 21:22:23.000000000 +0200
+++ mplayer-export-2008-09-03/configure	2008-10-16 21:30:47.000000000 +0200
@@ -589,7 +589,7 @@ _dvdnavconfig=dvdnav-config
 _dvdreadconfig=dvdread-config
 _dvdread=auto
 _dvdread_internal=auto
-_libdvdcss_internal=auto
+_libdvdcss_internal=no
 _xanim=auto
 _real=auto
 _live=auto
diff -up mplayer-export-2008-09-03/Makefile.nodvdcss mplayer-export-2008-09-03/Makefile
--- mplayer-export-2008-09-03/Makefile.nodvdcss	2008-10-16 21:22:23.000000000 +0200
+++ mplayer-export-2008-09-03/Makefile	2008-10-16 21:27:12.000000000 +0200
@@ -336,11 +336,6 @@ SRCS_COMMON-$(LIBDCA)                += 
 SRCS_COMMON-$(LIBDV)                 += libmpcodecs/ad_libdv.c \
                                         libmpcodecs/vd_libdv.c \
                                         libmpdemux/demux_rawdv.c
-SRCS_COMMON-$(LIBDVDCSS_INTERNAL)    += libdvdcss/css.c \
-                                        libdvdcss/device.c \
-                                        libdvdcss/error.c \
-                                        libdvdcss/ioctl.c \
-                                        libdvdcss/libdvdcss.c \
 
 SRCS_COMMON-$(LIBMAD)                += libmpcodecs/ad_libmad.c
 SRCS_COMMON-$(LIBMPEG2)              += libmpcodecs/vd_libmpeg2.c \
@@ -677,7 +672,6 @@ DIRS =  . \
         libavcodec/sparc \
         libavformat \
         libavutil \
-        libdvdcss \
         libdvdread \
         libfaad2 \
         libmenu \
@@ -783,7 +777,6 @@ codec-cfg.d: codecs.conf.h
 mencoder.d mplayer.d vobsub.d gui/win32/gui.d libmpdemux/muxer_avi.d osdep/mplayer-rc.o stream/network.d stream/stream_cddb.d: version.h
 $(DEPS): help_mp.h
 
-libdvdcss/%.o libdvdcss/%.d: CFLAGS += -D__USE_UNIX98 -D_GNU_SOURCE -DVERSION=\"1.2.9\" $(CFLAGS_LIBDVDCSS)
 libdvdread/%.o libdvdread/%.d: CFLAGS += -D__USE_UNIX98 -D_GNU_SOURCE -DHAVE_CONFIG_H $(CFLAGS_LIBDVDCSS_DVDREAD)
 libfaad2/%.o libfaad2/%.d: CFLAGS += -Ilibfaad2 -D_GNU_SOURCE -DHAVE_CONFIG_H $(CFLAGS_FAAD_FIXED)
 
